pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -391,36 +391,33 @@ def _map_tool_definition(f: ToolDefinition) -> ToolParam:
 def _map_usage(message: AnthropicMessage | RawMessageStreamEvent) -> usage.Usage:
     if isinstance(message, AnthropicMessage):
         response_usage = message.usage
+    elif isinstance(message, RawMessageStartEvent):
+        response_usage = message.message.usage
+    elif isinstance(message, RawMessageDeltaEvent):
+        response_usage = message.usage
     else:
-        if isinstance(message, RawMessageStartEvent):
-            response_usage = message.message.usage
-        elif isinstance(message, RawMessageDeltaEvent):
-            response_usage = message.usage
-        else:
-            # No usage information provided in:
-            # - RawMessageStopEvent
-            # - RawContentBlockStartEvent
-            # - RawContentBlockDeltaEvent
-            # - RawContentBlockStopEvent
-            response_usage = None
-
-    if response_usage is None:
+        # No usage information provided in:
+        # - RawMessageStopEvent
+        # - RawContentBlockStartEvent
+        # - RawContentBlockDeltaEvent
+        # - RawContentBlockStopEvent
         return usage.Usage()
 
-    # Store all integer-typed usage values in the details dict
-    response_usage_dict = response_usage.model_dump()
-    details: dict[str, int] = {}
-    for key, value in response_usage_dict.items():
-        if isinstance(value, int):
-            details[key] = value
+    # Store all integer-typed usage values in the details, except 'output_tokens' which is represented exactly by
+    # `response_tokens`
+    details: dict[str, int] = {
+        key: value
+        for key, value in response_usage.model_dump(exclude={'output_tokens'}).items()
```

Comment:
Is there any issue will leaving output_tokens in, even though it will be duplicated? It's very useful knowing downstream that we can rely on `details` to be a faithful representation of the vender-returned usage object. Ideally, it would also support floats too

Reply:
okay, I'll revert.

Reply:
done.